sdk/core/azure-core/azure/core/pipeline/transport/__init__.py

```diff
@@ -24,137 +24,92 @@
 
-        else:
-            try:
-                from ._aiohttp import AioHttpTransport, AioHttpTransportResponse
-                __all__.extend([
-                    'AioHttpTransport',
-                    'AioHttpTransportResponse',
-                ])
-            except ImportError:
-                pass  # Aiohttp not installed
-    except (ImportError, SyntaxError):
-        pass  # Asynchronous pipelines not supported.
+def __getattr__(name):
+    if name == 'AsyncioRequestsTransport':
+        try:
+            from ._requests_asyncio import AsyncioRequestsTransport
+            return AsyncioRequestsTransport
+        except ImportError:
+            pass
+    if name == 'AsyncioRequestsTransportResponse':
+        try:
+            from ._requests_asyncio import AsyncioRequestsTransportResponse
+            return AsyncioRequestsTransportResponse
+        except ImportError:
+            pass
+    if name == 'AsyncHttpTransport':
+        try:
+            from ._base_async import AsyncHttpTransport
+            return AsyncHttpTransport
+        except ImportError:
+            pass
+    if name == 'AsyncHttpResponse':
+        try:
+            from ._base_async import AsyncHttpResponse
+            return AsyncHttpResponse
+        except ImportError:
+            pass
+    if name == 'RequestsTransport':
+        try:
+            from ._requests_basic import RequestsTransport
+            return RequestsTransport
+        except ImportError:
+            raise ImportError("requests package is not installed")
+    if name == 'RequestsTransportResponse':
+        try:
+            from ._requests_basic import RequestsTransportResponse
+            return RequestsTransportResponse
+        except ImportError:
+            raise ImportError("requests package is not installed")
+    if name == 'AioHttpTransport':
+        try:
+            from ._aiohttp import AioHttpTransport
+            return AioHttpTransport
+        except ImportError:
+            raise ImportError("aiohttp package is not installed")
+    if name == 'AioHttpTransportResponse':
+        try:
+            from ._aiohttp import AioHttpTransportResponse
+            return AioHttpTransportResponse
+        except ImportError:
+            raise ImportError("aiohttp package is not installed")
+    if name == 'TrioRequestsTransport':
+        try:
+            from ._requests_trio import TrioRequestsTransport
+            return TrioRequestsTransport
+        except ImportError:
+            raise ImportError("trio package is not installed")
+    if name == 'TrioRequestsTransportResponse':
+        try:
+            from ._requests_trio import TrioRequestsTransportResponse
+            return TrioRequestsTransportResponse
+        except ImportError:
+            raise ImportError("trio package is not installed")
+    if name == '__bases__':
+        raise AttributeError("module 'azure.core.pipeline.transport' has no attribute '__bases__'")
+    return name
```

Comment:
Should this fail is the name don't match anything at all, or it's part of PEP562 to fail differently? IOW, what's the behavior of `from azure.core.pipeline.transport import NothingTransport` ? Does it return the string `NothingTransport` ?